nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-json-record-utils/src/main/java/org/apache/nifi/json/WriteJsonResult.java

```diff
@@ -165,19 +177,44 @@ public WriteResult writeRawRecord(final Record record) throws IOException {
         return WriteResult.of(incrementRecordCount(), attributes);
     }
 
+    private boolean isUseSerializeForm(final Record record, final RecordSchema writeSchema) {
+        final Optional<SerializedForm> serializedForm = record.getSerializedForm();
+        if (!serializedForm.isPresent()) {
+            return false;
+        }
+
+        final SerializedForm form = serializedForm.get();
+        if (!form.getMimeType().equals(getMimeType()) || !record.getSchema().equals(writeSchema)) {
+            return false;
+        }
+
+        final Object serialized = form.getSerialized();
+        String serializedString;
+
+        if (!(serialized instanceof String)) {
+            return false;
+        } else {
+            serializedString = (String) serialized;
+        }
+
```

Comment:
I feel it's risky to remove the check for pretty formatting form the backport. A later backport of https://issues.apache.org/jira/browse/NIFI-12480 is not straightforward, it's very possible this wouldn't be added back. I see no drawback in leaving it here.
```suggestion
        final boolean serializedPretty = serializedString.contains("\n");
        if (serializedPretty != this.prettyPrint) {
            return false;
        }
```

Reply:
@tpalfy Thank you for the comment, I agree, we need to avoid possible merge issues.
I think it would be the best if I backported [NIFI-12480](https://issues.apache.org/jira/browse/NIFI-12480) first and when it is merged to support/nifi-1.x, we can continue with the actual ticket. 
I created another ticket for backporting NIFI-12480, will create the PR for it soon: https://issues.apache.org/jira/browse/NIFI-13090